bindings/sftp/sftp.go

```diff
@@ -0,0 +1,322 @@
+package sftp
```

Comment:
Looks like there are some linter failures: https://github.com/dapr/components-contrib/actions/runs/10342318616/job/28666984066?pr=3505

Please run `make lint` locally to see the errors and address them!

Reply:
@ItalyPaleAle `G106: Use of ssh InsecureIgnoreHostKey should be audited`. Shouldn't we support this feature or bypass with `//nolint:gosec` ?

Reply:
I think `//nolint:gosec` is appropriate here! It's an option users must manually configure